packages/database/src/modelClasses/Entity.js

```diff
@@ -496,4 +505,69 @@ export class EntityModel extends MaterializedViewLogDatabaseModel {
 
     return level;
   }
+
+  async getEntityTypes() {
+    const entityTypes = await this.database.executeSql(
+      `SELECT UNNEST(enum_range(null::${ENTITY_TYPE_TYPE})) as type;`,
+    );
+    return entityTypes
+      .map(({ type }) => type)
+      .filter(type => !type.startsWith(DELETED_ENTITY_TYPE_PREFIX));
+  }
+
+  async addEntityType(name) {
+    // Have to manually format SQL here, as postgres doesn't support parameters on enum types
+    // Using pg-format to avoid SQL injection
```

Comment:
This one was a little tricky to deal with 😉 As described [here](https://github.com/brianc/node-postgres/issues/2259) postgres doesn't support parameter substitution for enum types. So went with the recommendation from others to use a 3rd party SQL injection parser.